mne_connectivity/spectral/epochs.py

```diff
@@ -89,14 +90,36 @@ def _prepare_connectivity(epoch_block, times_in, tmin, tmax,
     n_times = len(times)
 
     if indices is None:
-        logger.info('only using indices for lower-triangular matrix')
-        # only compute r for lower-triangular region
-        indices_use = np.tril_indices(n_signals, -1)
+        if any(this_method in _multivariate_methods for this_method in method):
+            if any(this_method in _gc_methods for this_method in method):
+                raise ValueError(
+                    'indices must be specified when computing Granger '
+                    'causality, as all-to-all connectivity is not supported')
+            else:
+                logger.info('using all indices for multivariate connectivity')
+                indices_use = (np.arange(n_signals, dtype=int),
+                               np.arange(n_signals, dtype=int))
+        else:
+            logger.info('only using indices for lower-triangular matrix')
+            # only compute r for lower-triangular region
+            indices_use = np.tril_indices(n_signals, -1)
     else:
+        if any(this_method in _gc_methods for this_method in method):
+            if set(indices[0]).intersection(indices[1]):
+                raise ValueError(
+                    'seed and target indices must not intersect when computing'
+                    'Granger causality')
         indices_use = check_indices(indices)
 
     # number of connectivities to compute
-    n_cons = len(indices_use[0])
+    if any(this_method in _multivariate_methods for this_method in method):
+        if len(np.unique(indices_use[0])) != len(np.unique(indices_use[1])):
+            raise ValueError(
```

Comment:
Can you add a unit-test for this case?

Reply:
Done! https://github.com/tsbinns/mne-connectivity/blob/894c42e59cef9f97aa07a7a4f71cef756512c50c/mne_connectivity/spectral/tests/test_spectral.py#L593-L599